app/models/hackathon/digest/listings/criterion.rb

```diff
@@ -0,0 +1,33 @@
+# Abstract base class for criteria that generate listings.
+class Hackathon::Digest::Listings::Criterion
+  def initialize(recipient:)
+    raise NotImplementedError, "Abstract base class" if instance_of?(Hackathon::Digest::Listings::Criterion)
+
+    @recipient = recipient
+  end
+
+  def candidate_listings
+    raise NotImplementedError
```

Comment:
This will never get reached, so maybe just show how it's meant to be used?

```suggestion
    # Hackathon.all
```

Reply:
Since this could still be reached by a subclass that doesn't implement `candidate_listings`, we'll keep it